src/basic/namespace-util.c

```diff
@@ -236,3 +236,29 @@ int userns_acquire(const char *uid_map, const char *gid_map) {
         return TAKE_FD(userns_fd);
 
 }
+
+int in_same_namespace(pid_t pid1, pid_t pid2, NamespaceType type) {
+        const struct namespace_info *nsinfo = &namespace_info[type];
+        char ns_path[STRLEN("proc//") + DECIMAL_STR_MAX(pid_t) + strlen(nsinfo->proc_path)];
+        struct stat ns_st1, ns_st2;
+
+        if (pid1 == 0)
+                pid1 = getpid_cached();
+
+        if (pid2 == 0)
+                pid2 = getpid_cached();
+
+        xsprintf(ns_path, "/proc/" PID_FMT "/%s", pid1, nsinfo->proc_path);
+        if (stat(ns_path, &ns_st1))
```

Comment:
Please use `if (stat(...) < 0)` even if it returns zero on success.

Reply:
ok